webroot/src/components/Users/UserList/UserList.ts

```diff
@@ -0,0 +1,284 @@
+        });
+    }
+
+    async handleSearch(): Promise<void> {
+        this.isFormLoading = true;
+
+        const { value: searchValue } = this.formData.userSearch;
+
+        if (searchValue) {
+            await this.fetchListData(searchValue);
+        }
+
+        this.isFormLoading = false;
+    }
+
+    async setDefaultSort(): Promise<void> {
+        const { listId } = this;
+        const defaultSortOption = this.sortOptions.find((option) => option.isDefault) || this.sortOptions[0];
+        const { option, direction } = this.sortingStore.sortingMap[listId] || {};
+
+        if (!option) {
+            await this.$store.dispatch('sorting/updateSortOption', {
+                sortingId: listId,
+                newOption: defaultSortOption.value,
+            });
+        }
+
+        if (!direction) {
+            await this.$store.dispatch('sorting/updateSortDirection', {
+                sortingId: listId,
+                newDirection: SortDirection.asc,
+            });
+        }
+    }
+
+    async setDefaultPaging(shouldForce = false): Promise<void> {
+        const { listId } = this;
+        const { page, size } = this.paginationStore.paginationMap[this.listId] || {};
+        const { prevLastKey } = this.usersStore;
+
+        if (!page || shouldForce) {
+            await this.$store.dispatch('pagination/updatePaginationPage', {
+                paginationId: listId,
+                newPage: DEFAULT_PAGE,
+            });
+        }
+
+        if (!size) {
+            await this.$store.dispatch('pagination/updatePaginationSize', {
+                paginationId: listId,
+                newSize: DEFAULT_PAGE_SIZE,
+            });
+        }
+
+        if (prevLastKey) {
+            this.prevKey = prevLastKey;
+        }
+    }
+
+    async fetchListData(search = ''): Promise<void> {
+        const sorting = this.sortingStore.sortingMap[this.listId];
+        const { option, direction } = sorting || {};
+        const pagination = this.paginationStore.paginationMap[this.listId];
+        const { page, size } = pagination || {};
+        const requestConfig: any = {};
+
+        if (option) {
+            const serverSortByMap = {
+                firstName: 'givenName',
+                lastName: 'familyName',
+                permissions: 'permissions',
+                affiliation: 'affiliation',
+                states: 'jurisdiction',
+                accountStatus: 'status',
+            };
+
+            requestConfig.sortBy = serverSortByMap[option];
+        }
+
+        if (direction) {
+            const serverSortDirectionMap = {
+                asc: 'ascending',
+                desc: 'descending',
+            };
+
+            requestConfig.sortDirection = serverSortDirectionMap[direction];
+        }
+
+        // Handle prev / next pages for server paging keys
+        if (page && !this.usersStore.error) {
+            if (this.nextKey && page !== 1) {
+                requestConfig.getNextPage = true;
```

Comment:
Shouldn't this be if `page` is not the last page?

Reply:
Got it. I'm still a little confused by page !== 1. What is that checking? (Is page 0- or 1-indexed?)

Reply:
`page` in this case is the page being requested. So if for some reason page 1 is being requested then we wouldn't send the `nextKey` to the server even if we had one stored.

Reply:
Got it. So I presume that means `page` is 1-indexed? @jsandoval81

Reply:
Yep!